instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -113,7 +113,7 @@ def get_host_port_url_tuple(scope):
     """
     server = scope.get("server") or ["0.0.0.0", 80]
     port = server[1]
-    server_host = server[0] + (":" + str(port) if port != 80 else "")
+    server_host = server[0] + (":" + str(port) if str(port) != "80" else "")
```

Comment:
I found this discrepancy in [Django's `AsyncRequestFactory`](https://github.com/django/django/blob/a948d9df394aafded78d72b1daa785a0abfeab48/django/test/client.py#L500) while testing, as the base scope is created with a port `'80'`, instead of using an integer.

I think that changing this check could avoid any issues with other potential callers using a string instead of an integer.